### PR TITLE
Refactors INullCoalescingExpression to ICoalesce expression, and prop…

### PR DESCRIPTION
**Pull request by 333fred**
…erties to sensible names. Fixes https://github.com/dotnet/roslyn/issues/21290. Tagging @dotnet/analyzer-ioperation @dotnet/roslyn-compiler for review.

### CODE REVIEW

**Comment by 333fred** [comment]
@dotnet/roslyn-infrastructure failure appears to be https://github.com/dotnet/roslyn/issues/21347.

**Comment by 333fred** [comment]
@dotnet-bot retest windows_release_unit64_prtest please

**Review by genlu — APPROVED** [review]
Consider adding a VB test for `ICoalesceExpression `

**Review by mavasani — APPROVED** [review]

**Review by cston — APPROVED** [review]

File: src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb [hunk 1]
```diff
@@ -286,7 +286,7 @@ Namespace Microsoft.CodeAnalysis.Semantics
         End Function
 
         Private Function CreateBoundMeReferenceOperation(boundMeReference As BoundMeReference) As IInstanceReferenceExpression
-            Dim instanceReferenceKind As InstanceReferenceKind = If(boundMeReference.WasCompilerGenerated, InstanceReferenceKind.Implicit, InstanceReferenceKind.Explicit)
+            Dim instanceReferenceKind As InstanceReferenceKind = If(boundMeReference.WasCompilerGenerated, instanceReferenceKind.Implicit, instanceReferenceKind.Explicit)
```
Comment (AlekseyTs) [inline src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb]: >instanceReferenceKind [](start = 107, length = 21)

This looks like an unintentional capitalization change.

Comment (AlekseyTs) [inline src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb]: >instanceReferenceKind [](start = 139, length = 21)

Here too

File: src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb [hunk 2]
```diff
@@ -1073,7 +1073,7 @@ Namespace Microsoft.CodeAnalysis.Semantics
 
         Private Function CreateBoundGotoStatementOperation(boundGotoStatement As BoundGotoStatement) As IBranchStatement
             Dim target As ILabelSymbol = boundGotoStatement.Label
-            Dim branchKind As BranchKind = BranchKind.GoTo
+            Dim branchKind As BranchKind = branchKind.GoTo
```
Comment (AlekseyTs) [inline src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb]: >branchKind [](start = 43, length = 10)

Here too.

File: src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb [hunk 3]
```diff
@@ -1083,7 +1083,7 @@ Namespace Microsoft.CodeAnalysis.Semantics
 
         Private Function CreateBoundContinueStatementOperation(boundContinueStatement As BoundContinueStatement) As IBranchStatement
             Dim target As ILabelSymbol = boundContinueStatement.Label
-            Dim branchKind As BranchKind = BranchKind.Continue
+            Dim branchKind As BranchKind = branchKind.Continue
```
Comment (AlekseyTs) [inline src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb]: >branchKind [](start = 43, length = 10)

Here too.

File: src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb [hunk 4]
```diff
@@ -1093,7 +1093,7 @@ Namespace Microsoft.CodeAnalysis.Semantics
 
         Private Function CreateBoundExitStatementOperation(boundExitStatement As BoundExitStatement) As IBranchStatement
             Dim target As ILabelSymbol = boundExitStatement.Label
-            Dim branchKind As BranchKind = BranchKind.Break
+            Dim branchKind As BranchKind = branchKind.Break
```
Comment (AlekseyTs) [inline src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb]: >branchKind [](start = 43, length = 10)

Here too.

**Comment by AlekseyTs** [comment]
Done with review pass.

File: src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb (hunk 1 above)
Comment (333fred) [inline src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb]: I seriously do not know why these keep changing.

**Comment by 333fred** [comment]
@AlekseyTs fixed.

**Review by AlekseyTs — APPROVED** [review]
LGTM

**Pull request merged**